be/src/exec/pipeline/exchange/exchange_sink_operator.cpp

```diff
@@ -711,24 +712,28 @@ Status ExchangeSinkOperator::serialize_chunk(const Chunk* src, ChunkPB* dst, boo
     return Status::OK();
 }
 
+// chunk_request should be copied if it will be sent by many times or its size >= config::brpc_max_body_size.
 int64_t ExchangeSinkOperator::construct_brpc_attachment(const PTransmitChunkParamsPtr& chunk_request,
-                                                        butil::IOBuf& attachment) {
+                                                        butil::IOBuf& attachment, bool copy) {
     int64_t attachment_physical_bytes = 0;
     for (int i = 0; i < chunk_request->chunks().size(); ++i) {
         auto chunk = chunk_request->mutable_chunks(i);
         chunk->set_data_size(chunk->data().size());
 
-        int64_t before_bytes = CurrentThread::current().get_consumed_bytes();
-        attachment.append(chunk->data());
-        attachment_physical_bytes += CurrentThread::current().get_consumed_bytes() - before_bytes;
+        if (copy) {
```

Comment:
Without attachment, I think brpc will have serialization/deserialization cost. You can check it. If so, the performance may have no improvement, even worse sometimes.

Reply:
you are right. I use append_user_data to avoid coping data in userspace, which is not need to be serialized also.